tensorflow_quantum/core/ops/math_ops/simulate_mps_test.py

```diff
@@ -0,0 +1,282 @@
+                                       ][:int(batch_size * 0.5)]))
+
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='do not match'):
+            # wrong symbol_values size.
+            simulate_mps.mps_1d_expectation(
+                util.convert_to_tensor(circuit_batch), symbol_names,
+                symbol_values_array[:int(batch_size * 0.5)],
+                util.convert_to_tensor([[x] for x in pauli_sums]))
+
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='cirq.Channel'):
+            # attempting to use noisy circuit.
+            noisy_circuit = cirq.Circuit(cirq.depolarize(0.3).on_each(*qubits))
+            simulate_mps.mps_1d_expectation(
+                util.convert_to_tensor([noisy_circuit for _ in pauli_sums]),
+                symbol_names, symbol_values_array,
+                util.convert_to_tensor([[x] for x in pauli_sums]))
+
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='not in 1D topology'):
+            # attempting to use a circuit not in 1D topology
+            # 0--1--2--3
+            #        \-4
+            circuit_not_1d = cirq.Circuit(
+                cirq.X(qubits[0])**sympy.Symbol(symbol_names[0]),
+                cirq.Z(qubits[1])**sympy.Symbol(symbol_names[0]),
+                cirq.CNOT(qubits[2], qubits[3]),
+                cirq.CNOT(qubits[2], qubits[4]),
+            )
+            simulate_mps.mps_1d_expectation(
+                util.convert_to_tensor([circuit_not_1d for _ in pauli_sums]),
+                symbol_names, symbol_values_array,
+                util.convert_to_tensor([[x] for x in pauli_sums]))
+
+        with self.assertRaisesRegex(tf.errors.InvalidArgumentError,
+                                    expected_regex='not in 1D topology'):
+            # attempting to use a circuit in 1D topology, which looks in 2D.
+            # 0--1
+            #  \-2-\
+            #    3--4  == 1--0--2--4--3
+            circuit_not_1d = cirq.Circuit(
+                cirq.CNOT(qubits[0], qubits[1]),
+                cirq.CNOT(qubits[0], qubits[2]),
+                cirq.CNOT(qubits[2], qubits[4]),
+                cirq.CNOT(qubits[3], qubits[4]),
+            )
+            simulate_mps.mps_1d_expectation(
+                util.convert_to_tensor([circuit_not_1d for _ in pauli_sums]),
+                symbol_names, symbol_values_array,
+                util.convert_to_tensor([[x] for x in pauli_sums]))
+
+        res = simulate_mps.mps_1d_expectation(
+            util.convert_to_tensor([cirq.Circuit() for _ in pauli_sums]),
+            symbol_names, symbol_values_array.astype(np.float64),
+            util.convert_to_tensor([[x] for x in pauli_sums]))
+        self.assertDTypeEqual(res, np.float32)
+
+    def test_simulate_mps_1d_expectation_results(self):
```

Comment:
This is a good start. Could we add a few more tests here that do the following:

1. Test with all single qubit gates in `tfq.get_supported_gates`
2. Test with all single qubit gates in `tfq.get_supported_gates.controlled_by(a_qubit)` and ensure that this fails or passes accordingly)
3. Test that all two qubit gates in `tfq.get_supported_gates` works.
4. Ensure that the empty case works as expected